sequencing-server/src/lib/mustache/util/time.ts

```diff
@@ -0,0 +1,313 @@
+// TODO: Ensure absolute and relative times resolve correctly. Ask about this later.
+
+import { ParsedDoyString, ParsedDurationString, ParsedYmdString, TimeTypes } from './types/time.js';
+import { Temporal } from '@js-temporal/polyfill';
+import { SequencingLanguage } from '../enums/language.js';
+
+
+/////////////// SEQUENCING-SERVER-SPECIFIC HELPERS ///////////////
+export function addTime(startTime: string, duration: string, environment: { language: SequencingLanguage }): string {
+  let date: Temporal.Instant
+  if (environment.language === SequencingLanguage.STOL) {
+    date = STOLToISO8061(startTime)
+  }
+  else { // Text and SeqN are handled the same.
+    date = SeqNToISO8061(startTime)
+  }
+
+  let dur: Temporal.Duration;
+  if (duration.includes(":")) {
+    dur = Temporal.Duration.from(AERIEDurationToISO8061(duration))
+  }
+  else {
+    dur = Temporal.Duration.from(duration)
+  }
+  date = date.add(dur)
+
+  if (environment.language === SequencingLanguage.STOL) {
+    return ISO8061toSTOL(date)
+  }
+  else { // Text and SeqN are handled the same.
+    return ISO8061toSeqN(date)
+  }
+}
+
+export function subtractTime(startTime: string, duration: string, environment: { language: SequencingLanguage }): string {
+  let date: Temporal.Instant
+  if (environment.language === SequencingLanguage.STOL) {
+    date = STOLToISO8061(startTime)
+  }
+  else { // Text and SeqN are handled the same.
+    date = SeqNToISO8061(startTime)
+  }
+
+  let dur: Temporal.Duration;
+  if (duration.includes(":")) {
+    dur = Temporal.Duration.from(AERIEDurationToISO8061(duration))
+  }
+  else {
+    dur = Temporal.Duration.from(duration)
+  }
+  date = date.subtract(dur)
+
+  if (environment.language === SequencingLanguage.STOL) {
+    return ISO8061toSTOL(date)
+  }
+  else { // Text and SeqN are handled the same.
+    return ISO8061toSeqN(date)
+  }
+
+}
+
+// TIME PARSING
+export function SeqNToISO8061(date: string): Temporal.Instant {
+  return Temporal.Instant.from(convertDoyToYmd(date))
+}
+
+export function STOLToISO8061(date: string): Temporal.Instant {
+  return Temporal.Instant.from(convertDoyToYmd(date))
+}
+
+function checkNumDurationComponents(split: string[]): split is [string, string, string] {
+  return split.length === 3;
+}
+
+export function AERIEDurationToISO8061(duration: string): string {
+  // HHHHHH...:MM:SS.mmmuuu -> PHHMMSS.mmmuuuS
+  let split = duration.split(":")
+
+  if (!checkNumDurationComponents(split)) {
```

Comment:
This method can be inlined for clarity.
```suggestion
  if (split.length !== 3) {
```

Reply:
It seems that the current notation was required so our editors wouldn't flag a type error. Inlining it leads the type checking system to believe that `split[0]` is `string|undefined`, but as is, it believes `split` is exactly `[string, string, string]`.